src/common/SConscript

```diff
@@ -22,14 +22,24 @@ def scons():
 
     prereqs.require(denv, 'protobufc')
     denv.AppendUnique(LIBS=['cart', 'gurt'])
+
+    # TODO: need add qat dependency
+    #denv.AppendUnique(CPPDEFINES =['-DHAVE_QAT'])
+    #denv.AppendUnique(LIBS=['qat_s'])
+    #denv.AppendUnique(CPPPATH=['/home/wli5/qat/quickassist/include/',
+    #    '/home/wli5/qat/quickassist/include/lac', 
+    #    '/home/wli5/qat/quickassist/include/dc',
+    #    '/home/wli5/qat/quickassist/lookaside/access_layer/include', 
+    #    '/home/wli5/qat/quickassist/utilities/libusdm_drv/'])
```

Comment:
@jolivier23 Would you have any suggestions on conditionally support QaT? The idea is to detect whether it is installed on the node and enabled QAT support if so?
The other approach would be to add QAT build into scons_local, but we will still need to disable QAT when building RPMs since we don't want to package QAT ourselves (the QAT team is working on RPMs, but it is not available yet).
Any suggestions?

Reply:
Please add the following:

In utils/sl/components/__init__.py, copy one of the existing definitions...for instance this one

reqs.define('uuid', libs=['uuid'], headers=['uuid/uuid.h'],
227                     package='libuuid-devel')

with something like

reqs.define('qat', libs=list_of_ilbs, headers=list_of_headers, package='qat_nonexist') # I understand there isn't an RPM package for this but I think the package name is required for components we don't build ourselves)

Then in your SConscript file you can do this

if prereqs.check_component('qat'):
    prereqs.require(denv, 'qat')
    denv.AppendUnique(.....)

This will check to see if qat is available and conditionally compile this code.   Ideally, we would always compile it and have a runtime check but this is a reasonable first step

Reply:
Since you have it in a custom location though, what we may want to do here is have you make use of a ~/.scons_localrc file to setup CPPPATH.  I presume if they were installed in system default, none of that would be needed because it would be taken care of by things like ldconfig.

That file, if it exists, gets executed before these checks so you could simply set your paths as follows

Import('env')
env.AppendUnique(....)

the LIBS would still go in the SConscript

Reply:
Thanks for the advice. But I'm really not familiar with SConscript, if you can help on that, it will be great, it could be in another PR.